packages/didcomm/src/modules/connections/__tests__/connection-manual.test.ts

```diff
@@ -147,9 +170,82 @@ describe('Manual Connection Flow', () => {
     expect(bobConnectionRecord).toBeConnectedWith(faberBobConnectionRecord)
+  })
 
-    await aliceAgent.shutdown()
-    await bobAgent.shutdown()
-    await faberAgent.shutdown()
+  it.only('allows a party to decline an incoming connection request', async () => {
+    const aliceSpy = jest.spyOn(aliceMessageSender, 'sendMessage')
+    const bobSpy = jest.spyOn(bobMessageReceiver, 'receiveMessage')
+    const bobSpy2 = jest.spyOn(bobMessageReceiver as any, 'transformAndValidate')
+    // const bobConnectionService = bobAgent.dependencyManager.resolve(ConnectionService)
+    // const bobConnectionSpy = jest.spyOn(bobConnectionService, 'processProblemReport')
+
+    // Alice creates an OOB invitation
+    const aliceOutOfBandRecord: OutOfBandRecord = await aliceAgent.modules.oob.createInvitation({
+      autoAcceptConnection: false,
+    })
+
+    // Bob receives the invitation and initiates a connection
+    const {
+      outOfBandRecord: bobOutOfBandRecord,
+      connectionRecord: bobConnectionRecord,
+    } = await bobAgent.modules.oob.receiveInvitation(
+      aliceOutOfBandRecord.outOfBandInvitation,
+        {
+          autoAcceptInvitation: true,
+          autoAcceptConnection: false,
+        }
+      )
+
+    // Bob's internal state should reflect waiting for a response
+    expect(bobOutOfBandRecord.state).toBe(OutOfBandState.PrepareResponse)
+    expect(bobConnectionRecord.state).toBe(DidExchangeState.RequestSent)
+
+    // Alice should receive the connection request and her state should reflect this
+    const aliceConnectionRecord = await waitForRequest(aliceAgent, 'bob')
+    expect(aliceConnectionRecord.state).toBe(DidExchangeState.RequestReceived)
+
+    // Now Alice declines the request
+    const aliceConnectionAfterDecline = await aliceAgent.modules.connections.declineRequest(aliceConnectionRecord.id)
+
+    // Alice should send the problem report correctly
+    expect(aliceSpy).toHaveBeenCalled()
+    const [aliceMessage] = aliceSpy.mock.calls[0]
+    expect(aliceMessage.message.type).toBe('https://didcomm.org/connection/1.0/problem-report')
+    expect(aliceMessage.message).toMatchObject({
+      description: { en: expect.stringContaining('Connection request declined') }
+    })
+
+    // Alice's agent reflects this in her OOB record and connection record
+    expect(aliceConnectionAfterDecline.state).toBe(DidExchangeState.Abandoned)
+    const aliceOobAfterDecline = await aliceAgent.modules.oob.findById(aliceOutOfBandRecord.id)
+    expect(aliceOobAfterDecline.state).toBe(OutOfBandState.Done)
+
+    // Bob should receive the problem report correctly
+    expect(bobSpy).toHaveBeenCalled()
+    const [bobMessage] = bobSpy.mock.calls[0]
+
+    expect(bobSpy2).toHaveBeenCalled()
+    const [spy2] = bobSpy2.mock.calls
+    // expect(bobConnectionSpy).toHaveBeenCalled()
+    // const [spyMessage] = bobConnectionSpy.mock.calls[0]
+    
+    console.debug(aliceMessage)
+    console.log(bobMessage)
+    console.log(spy2)
```

Comment:
This needs to be clean up. I am curious, how is the decline request currently handled by Bob? Since it should not allow the problem report to be assocaited with the invitation/connection, since we can't verify whether the sender of the problem report is the inviter based on the current code in declineRequest

Reply:
This was purely WIP, but I agree the problem report must contain something known to both Alice and Bob which would either be the `invitationId` from the OOB record, or the `threadId` from the connectionRecord. I set the threadId of the problem report in the `declineRequest` method to equal that of the connectionRecord.